### PR TITLE
add daskhub example

### PR DESCRIPTION
**Pull request by nabuskey**
### What does this PR do?

Adds an example composition for daskhub deployment.

### CODE REVIEW

File: examples/aws-provider/composite-resources/daskhub/example-dask-hub-password.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  compositionSelector:
+    matchLabels:
+      awsblueprints.io/configuration: password
+      awsblueprints.io/environment: test
+  resourceConfig:
+    awsProviderConfigName: aws-provider-config
+    kubernetesProviderConfigName: example-cas
+    region: us-west-2
+  parameters:
+    eksOidc: oidc.eks.us-west-2.amazonaws.com/id/123ABCDIEKEKEKEF1BK
```
Comment (candonov) [inline examples/aws-provider/composite-resources/daskhub/example-dask-hub-password.yaml:20]: Can this value be created dynamically in the composition, or be referenced as "observe-only" resource or added to enviromentConfig?

**Review by candonov — COMMENTED** [review]
Can you add a README to this example?

File: compositions/aws-provider/daskhub/test-env.yaml [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+                      service:
+                        type: ClusterIP
+                    cull:
+                      enabled: true
+                      timeout: 3600
+                      every: 300
+                    hub:
+                      services:
+                        dask-gateway:
+                          apiToken: '36b06f3dcb40d7344aa08c1189105ec96979696658dcc7dc2459ce0f224c4c66'
```
[the remark this answers is not in the document]
Comment (nabuskey) [reply compositions/aws-provider/daskhub/test-env.yaml:69]: These are not too sensitive. Only usable within the cluster for this deployment.
I should add an README and document this though.

File: compositions/aws-provider/daskhub/test-env.yaml [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+                    enabled: true  # Enabling dask-gateway will install Dask Gateway as a dependency.
+                    # Futher Dask Gateway configuration goes here
+                    # See https://github.com/dask/dask-gateway/blob/master/resources/helm/dask-gateway/values.yaml
+                    gateway:
+                      prefix: "/services/dask-gateway"  # Users connect to the Gateway through the JupyterHub service.
+                      auth:
+                        type: jupyterhub  # Use JupyterHub to authenticate with Dask Gateway
+                        jupyterhub:
+                          apiToken: '36b06f3dcb40d7344aa08c1189105ec96979696658dcc7dc2459ce0f224c4c66'
+                          # apiUrl: http://ac7f0b1a7bf8e4cb89f7a12a5b94796b-323918181.us-west-2.elb.amazonaws.com/hub/api
```
[the remark this answers is not in the document]
Comment (nabuskey) [reply compositions/aws-provider/daskhub/test-env.yaml:131]: I need to fix it. Thanks!

File: examples/aws-provider/composite-resources/daskhub/example-dask-hub-password.yaml (hunk 1 above)
Comment (nabuskey) [reply examples/aws-provider/composite-resources/daskhub/example-dask-hub-password.yaml:20]: Yes, possible to enable that within the composition. It's meant to be usable by just specifying these required parameters.

**Comment by nabuskey** [comment]
I will add a README file for this.

File: compositions/aws-provider/daskhub/test-env.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (candonov) [reply compositions/aws-provider/daskhub/test-env.yaml:69]: Even if not "too" sensitive, it should not be in github. One option would be to add in the readme instructions on how to create this api token, kubectl command to add it to a k8s secrets and patch from the secret.

[the remark this answers is not in the document]
Comment (nabuskey) [reply compositions/aws-provider/daskhub/test-env.yaml:69]: You are right. Let me work on that.

**Comment by nabuskey** [comment]
Added readme and removed hardcoded password and api tokens.

File: examples/aws-provider/composite-resources/daskhub/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# DaskHub composition
+
+## DaskHub with simple password authentication
+
+This example creates a DaskHub installation with simple password authentication. This is suitable for testing environments only. Use a proper [supported authentication methods](https://jupyterhub.readthedocs.io/en/stable/tutorial/getting-started/authenticators-users-basics.html) for your production environments.
```
Comment (candonov) [inline examples/aws-provider/composite-resources/daskhub/README.md:5]:
```suggestion
This example creates a DaskHub installation with simple password authentication which is suitable for testing environments only. To set up authentication in production environments, refer to the JupiterHub documentation on [supported authentication methods](https://jupyterhub.readthedocs.io/en/stable/tutorial/getting-started/authenticators-users-basics.html). 
```

File: examples/aws-provider/composite-resources/daskhub/README.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+1. Run the following command to create a API token that is used by Dask Gateway and Jupyterhub to talk to each other. This token is also used as user password.
+
+```bash
+sed "s/REPLACEME/$(openssl rand -hex 32)/g" examples/aws-provider/composite-resources/daskhub/daskhub-sensitive.yaml | kubectl apply -f -
+```
+
+2. Get your EKS cluster's OIDC ID. See [this documentation](https://docs.aws.amazon.com/eks/latest/userguide/associate-service-account-role.html) for more information. Be sure to replace `my-cluster` with your cluster name in the command.
+
+```bash
+oidc_provider=$(aws eks describe-cluster --name my-cluster --region $AWS_REGION --query "cluster.identity.oidc.issuer" --output text | sed -e "s/^https:\/\///")
```
Comment (candonov) [inline examples/aws-provider/composite-resources/daskhub/README.md:16]: Add a step to
```bash
export AWS_REGION=<aws_region>
```
possibly also
```bash
export CLUSTER=<my_cluster>
```

File: examples/aws-provider/composite-resources/daskhub/README.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+5. The default public service is disabled. We need to forward connection to access the UI.
+
+```bash
+# port forward
+kubectl port-forward -n daskhub svc/proxy-public  8080:80
+# get your login password
+get secret daskhub-sensitive-values --template='{{ index .data "config.yaml" | base64decode}}' | yq '.jupyterhub.hub.services.dask-gateway.apiToken'
+```
+
+6. Open your web browser and navigate to [`http://localhost:8080`](http://localhost:8080). You should be able to login with the password obtained above.
```
Comment (candonov) [inline examples/aws-provider/composite-resources/daskhub/README.md:40]:
```suggestion
6. Open your web browser and navigate to [`http://localhost:8080`](http://localhost:8080). Login with the password obtained in previous step.
```

File: examples/aws-provider/composite-resources/daskhub/README.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
+# DaskHub composition
+
```
Comment (candonov) [inline examples/aws-provider/composite-resources/daskhub/README.md:2]: Can you provide a description of what the purpose of this example is?

**Review by candonov — APPROVED** [review]
lgtm

**Pull request merged**